### PR TITLE
RWA-1035 - Entity inserted into DB should update delivery count on conflict

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1035

### Change description ###

LLD mentions that delivery count of the entity should be incremented if initial save of entity errors/conflicts.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by rafalkalita — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java [hunk 1]
```diff
@@ -87,6 +87,20 @@ public CaseEventMessage getMessage(String messageId) {
         return mapper.mapToCaseEventMessage(messageEntity);
     }
 
+    private CaseEventMessageEntity insertMessage(CaseEventMessageEntity caseEventMessageEntity) {
+
+        final List<CaseEventMessageEntity> existingMessages =
+                repository.findByMessageId(caseEventMessageEntity.getMessageId());
+        if (existingMessages.isEmpty()) {
+            return repository.save(caseEventMessageEntity);
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java:95]: still default? should this be using native sql as specified in the story description?

**Review by BenNzewi — COMMENTED** [review]
we don't seem to be covering these error scenarios? Is this part of a separate ticket?
https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1531414005#WA:HandlingCaseEventsInOrderProcessingandDLQs-ErrorScenarios

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java (hunk 1 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java:95]: Was so much more difficult to code with native SQL query, requiring casts of json data etc - this is much simpler and is functionally equivalent

**Comment by sullivangareth** [comment]
> we don't seem to be covering these error scenarios? Is this part of a separate ticket? https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1531414005#WA:HandlingCaseEventsInOrderProcessingandDLQs-ErrorScenarios

Yes - the error scenarios are to be implemented in https://tools.hmcts.net/jira/browse/RWA-975

**Pull request merged**